cmd/forwarder/fixer.go

```diff
@@ -21,48 +23,69 @@ const (
 	maxMsgidLength    = 32
 )
 
-var nilVal = []byte(`- `)
-var queryParams = []string{"index", "source", "sourcetype"}
-
 // Get metadata from the http request.
 // Returns an empty byte array if there isn't any.
 func getMetadata(req *http.Request, cred *credential, metadataId string) ([]byte, bool) {
 	var metadataWriter bytes.Buffer
 	var foundMetadata bool
+
 	// Calculate metadata query parameters
 	if metadataId != "" {
+		var queryParams = []string{"index", "source", "sourcetype"}
+		var fieldsBuilder strings.Builder
+
+		// Pre-grow to typical lengths to avoid extra slice allocations
+		fieldsBuilder.Grow(128)
+		metadataWriter.Grow(1024)
+
+		queryFieldParams := strings.Split(os.Getenv("LOG_ISS_FIELD_PARAMS"), ",")
+		queryParams = append(queryParams, queryFieldParams...)
```

Comment:
@scottso suggestion:  Instead of putting `os.Getenv` here, we should pass the params into this method. We should use config.go to get values from ENV, here: https://github.com/heroku/log-iss/blob/e1e7ede05228f48615e4ac16891489bf4a1bd3e9/cmd/forwarder/config.go#L16